### ISSUE TITLE
`prefer-lookaround` should ignore `^` and `$`

### ISSUE THREAD

**Issue by fisker** [ota-meshi/eslint-plugin-regexp#478]
**Information:**
- ESLint version: 8.27.0
- `eslint-plugin-regexp` version: 7.31.10

**Description**

```js
var str = 'JavaScript'.replace(/Java(Script)/g, 'Type$1')
var str = 'JavaScript'.replace(/Java(Script$)/g, 'Type$1')
var str = 'JavaScript'.replace(/Java(Script)$/g, 'Type$1')
```

The first two are reported, but the 3rd is not.

I think 

```js
var str = 'JavaScript'.replace(/Java(?=Script)$/g, 'Type')
```

should result the same. Will adding support for this case break things?

**Comment by fisker** [comment]
> It's not, the assertion after the capturing group has to be placed at the end of the new lookahead. 

Thank you! You are right.

**Comment by fisker** [comment]
So, to summarize

```js
var str = 'JavaScript'.replace(/Java(Script)$/g, 'Type$1')

// ->

var str = 'JavaScript'.replace(/Java(?=Script$)/g, 'Type')
```

And

```js
var str = 'JavaScript'.replace(/Java(S)(c)(r)(i)(p)(t)/g, 'Type$1$2$3$4$5$6')

// ->

var str = 'JavaScript'.replace(/Java(S)(c)(r)(i)(p)(?=t)/g, 'Type$1$2$3$4$5')

// ->

var str = 'JavaScript'.replace(/Java(S)(c)(r)(i)(?=pt)/g, 'Type$1$2$3$4')

// ...

var str = 'JavaScript'.replace(/Java(?=Script)/g, 'Type')
```
